polkadot/runtime/parachains/src/disputes/slashing.rs

```diff
@@ -274,18 +270,33 @@ where
 		losers: impl IntoIterator<Item = ValidatorIndex>,
 		backers: impl IntoIterator<Item = ValidatorIndex>,
 	) {
-		let kind = SlashingOffenceKind::ForInvalid;
-		Self::do_punish(session_index, candidate_hash, kind, losers, backers);
+		let losers: Vec<_> = losers.into_iter().collect();
+		let backers: BTreeSet<_> = backers.into_iter().collect();
```

Comment:
Is there a particular reason why we want to use `BTreeSet`  and not `HashSet`? We don't seem to need the ordering and also attacker would not be able to control the hash collisions to slow the runtime down.

Reply:
is this about effiency in the context of https://github.com/paritytech/polkadot-sdk/pull/8606? i doubt this is relevant here as the size of backers is limited by 5 (and for losers by 300) - we could even use sorted vecs, but using btreesets is type safer